cmd/extract.go

```diff
@@ -34,51 +72,61 @@ func extractNVC(arcPath string, pathlist []string, outputDirectory string, extra
 	for hash, nvcEntry := range entries {
 		_, exists := hashToPath[hash]
 		if extractUnknown || exists {
-			data, err := nvcEntry.Data(r)
+			data, err := nvcEntry.Data(arcFile)
 			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
 				continue
 			}
 
-			var path string
+			var path string // File path/name as stored (hashed) in ToC
 
-			if exists {
+			if exists { // If this entry's name was in pathlist, then we use its name...
 				path = hashToPath[hash]
-			} else {
+			} else { // ...And if it wasn't, then we make up a name using the hash
 				var (
 					dirName string
 					ext     string
 				)
 
 				switch string(data[:4]) {
 				case "\x89PNG":
-					dirName = "data/unknown_png/"
+					dirName = filepath.Join("data", "unknown_png")
 					ext = ".png"
 				case "nmf1":
-					dirName = "data/unknown_nmd/"
+					dirName = filepath.Join("data", "unknown_nmd")
 					ext = ".nmd"
 				case "OggS":
-					dirName = "data/unknown_ogg/"
+					dirName = filepath.Join("data", "unknown_ogg")
 					ext = ".ogg"
 				case "RIFF":
-					dirName = "data/unknown_wav/"
+					dirName = filepath.Join("data", "unknown_wav")
 					ext = ".wav"
 				default:
-					dirName = "data/unknown/"
+					dirName = filepath.Join("data", "unknown")
 					ext = ".unknown"
 				}
 
-				path = fmt.Sprintf("%s%v%s", dirName, hash, ext)
+				path = filepath.Join(dirName, hash.String()+ext)
 			}
 
-			outputPath := outputDirectory + path
+			outputPath := filepath.Join(outputDirectory, path)
```

Comment:
Can we add a `-v` flag, make it log extracted paths?

Reply:
Yeah, that's a decent idea. Log those messages to stdout and leave the error messages on stderr so that they can be redirected separately, I guess?

Reply:
Yep that sounds good.  paths to stdout, warnings/errors to stderr

Reply:
Added in d4e4268a4391a28043458c47ae0697f23d520204:

```go
if verbose {
    fmt.Printf("Extracting %s to %s\n", hash.String(), outputPath)
}
```

Reply:
i queued up a suggestion to simply print te path